vm/transaction.go

```diff
@@ -37,6 +38,7 @@ func marshalTxn(txn core.Transaction) (json.RawMessage, error) {
 		txnAndQueryBit.Txn["Declare"] = map[string]any{
 			"V" + t.Version.Text(felt.Base10): t,
 		}
+	// why stored value in Txn is different for these two ?
```

Comment:
That is due to how the object on Rust side expects the JSON to be formatted. take a look at Rust enums.